airflow/models/baseoperator.py

```diff
@@ -1800,6 +1801,53 @@ def wait_for_downstream(self) -> bool:
     def depends_on_past(self) -> bool:
         return self.partial_kwargs.get("depends_on_past") or self.wait_for_downstream
 
+    def expand_mapped_task(self, upstream_ti: "TaskInstance", session: "Session" = NEW_SESSION) -> None:
+        """Create the mapped TaskInstances for mapped task."""
+        # TODO: support having multiuple mapped upstreams?
+        from airflow.models.taskmap import TaskMap
+        from airflow.settings import task_instance_mutation_hook
+
+        task_map_info: TaskMap = (
+            session.query(TaskMap)
+            .filter_by(
+                dag_id=upstream_ti.dag_id,
+                task_id=upstream_ti.task_id,
+                run_id=upstream_ti.run_id,
+                map_index=upstream_ti.map_index,
+            )
+            .one()
+        )
+
+        unmapped_ti: Optional[TaskInstance] = upstream_ti.dag_run.get_task_instance(
+            self.task_id, map_index=-1, session=session
+        )
+
+        maps = range(task_map_info.length)
+
+        if unmapped_ti:
+            # The unmapped TaskInstance still exists -- this means we haven't
+            # tried to run it before.
+            unmapped_ti.map_index = 0
+            maps = range(1, task_map_info.length)
```

Comment:
There’s an edge case I just realised, what should happen if the upstream returns an empty list/dict? It feels wrong failing the upstream, but what should we do to keep the graphs making sense? Keep the unmapped TI, and mark it as SKIPPED? Delete it without expanding? Other possibilities?

Reply:
I think skipping makes sense, yeah.

Reply:
I pushed a change to mark the unmapped ti as SKIPPED in this case.

Reply:
I pushed a change to mark the unmapped ti as SKIPPED in this case.